presto-hive/src/test/java/com/facebook/presto/hive/HiveTestingEnvironment.java

```diff
@@ -0,0 +1,187 @@
+    {
+        this.hdfsEnvironment = hdfsEnvironment;
+        this.locationService = locationService;
+        this.metadata = metadata;
+        this.metastoreClient = metastoreClient;
+        this.pageSinkProvider = pageSinkProvider;
+        this.pageSourceProvider = pageSourceProvider;
+        this.splitManager = splitManager;
+        this.executor = executor;
+        this.timeZone = timeZone;
+    }
+
+    public void close()
+    {
+        if (executor != null) {
+            executor.shutdownNow();
+            executor = null;
+        }
+    }
+
+    public ConnectorMetadata getMetadata()
+    {
+        return metadata;
+    }
+
+    public HdfsEnvironment getHdfsEnvironment()
+    {
+        return hdfsEnvironment;
+    }
+
+    public LocationService getLocationService()
+    {
+        return locationService;
+    }
+
+    public HiveMetastore getMetastoreClient()
+    {
+        return metastoreClient;
+    }
+
+    public ConnectorPageSinkProvider getPageSinkProvider()
+    {
+        return pageSinkProvider;
+    }
+
+    public ConnectorPageSourceProvider getPageSourceProvider()
+    {
+        return pageSourceProvider;
+    }
+
+    public ConnectorSplitManager getSplitManager()
+    {
+        return splitManager;
+    }
+
+    public DateTimeZone getTimeZone()
+    {
+        return timeZone;
+    }
+
+    public static HiveTestingEnvironment newInstance(String host, int port, String timeZoneId)
+    {
+        ExecutorService executor = newCachedThreadPool(daemonThreadsNamed("hive-%s"));
+
+        HiveClientConfig hiveClientConfig = new HiveClientConfig();
+        hiveClientConfig.setTimeZone(timeZoneId);
+        String proxy = System.getProperty("hive.metastore.thrift.client.socks-proxy");
+        if (proxy != null) {
+            hiveClientConfig.setMetastoreSocksProxy(HostAndPort.fromString(proxy));
+        }
+
+        HiveCluster hiveCluster = new TestingHiveCluster(hiveClientConfig, host, port);
+        HiveMetastore metastoreClient = new CachingHiveMetastore(hiveCluster, executor, Duration.valueOf("1m"), Duration.valueOf("15s"));
+        HiveConnectorId connectorId = new HiveConnectorId("hive-test");
+        HdfsConfiguration hdfsConfiguration = new HiveHdfsConfiguration(new HdfsConfigurationUpdater(hiveClientConfig));
+
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(hdfsConfiguration, hiveClientConfig);
+        LocationService locationService = new HiveLocationService(metastoreClient, hdfsEnvironment);
+        JsonCodec<PartitionUpdate> partitionUpdateCodec = JsonCodec.jsonCodec(PartitionUpdate.class);
+        DateTimeZone timeZone = DateTimeZone.forTimeZone(TimeZone.getTimeZone(timeZoneId));
+        ConnectorMetadata metadata = new HiveMetadata(
+                connectorId,
+                metastoreClient,
+                hdfsEnvironment,
+                new HivePartitionManager(connectorId, hiveClientConfig),
+                timeZone,
+                10,
```

Comment:
I don't know if you wrote this code or just copied it over, but it is weird that this one uses hard coded values for config, but the one below gets the config from the config class.  Also it would be simpler if we just setup a config instance (so we can see the names of the fields), and pass them into `HiveMetadata` and `HiveSplitManager`